apiserver/facades/controller/caasoperatorprovisioner/provisioner.go

```diff
@@ -134,26 +134,35 @@ func (a *API) OperatorProvisioningInfo(args params.Entities) (params.OperatorPro
 		modelConfig,
 	)
 
-	imageRepo, err := docker.NewImageRepoDetails(cfg.CAASImageRepo())
+	imageRepo, exists := modelConfig.CAASImageRepo()
+	if !exists {
+		imageRepo = controller.CAASImageRepo
+
+		if imageRepo == "" {
+			imageRepo = podcfg.JujudOCINamespace
+		}
+	}
+
+	imageRepoDetails, err := docker.NewImageRepoDetails(imageRepo)
```

Comment:
We can refactor this to make it cleaner - here and elsewhere.
The `GetJujuOCIImagePath` method calls `docker.NewImageRepoDetails`  and uses the details result internally. We can just return that from the `GetOCIImagePath` call (maybe even rename that method to `GetJujuOCIImageDetails` or something. Doing this would avoid the extra block of code above being copied at every call site.